monitoring/api/v3/alerts-client/noxfile.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import os
+from pathlib import Path
+
+import nox
+
+# Proposed TEST_CONFIG.
+# You can have a file `testconfig` or something and it will be injected into
+# the generated noxfile.py.
+TEST_CONFIG = {
+    'use_multi_project': True,
```

Comment:
Rather than `use_multi_project: bool`, could we make this something like `override_project_variable: str` and pass in an env var to reference instead?

So something like:
```python
TEST_CONFIG = {
    `override_project_variable`: "VERSION_PYTHON_SAMPLES_TEST",
    # ...
} 

# ...
ret = {}
if TEST_CONFIG.get('override_project_variable'):
    ret[GOOGLE_CLOUD_PROJECT] = os.environ.get(TEST_CONFIG['override_project_variable'])

```

I think something like this might be preferable because:
1. We can customize the project per build, and don't have to regenerate all of the noxfiles when we make a change
2. It lets us keep the project id used private

Reply:
Good call for using an envvar.

Maybe we can just use the an envvar `BUILD_SPECIFIC_GCLOUD_PROJECT`, and keep the config simple.

```python
TEST_CONFIG = {
    'use_multi_project': True,
    ...
}

MULTI_PROJECT_OVERRIDE = "BUILD_SPECIFIC_GCLOUD_PROJECT"

    # then use it when the env var exists.
    if TEST_CONFIG.get('use_multi_project'):
        ret[GOOGLE_CLOUD_PROJECT] = os.environ.get(
            MULTI_PROJECT_OVERRIDE, os.environ.get('GOOGLE_CLOUD_PROJECT'))

```

Unless we want to change that environment variable name (`BUILD_SPECIFIC_GCLOUD_PROJECT`).

Reply:
Maybe config key should be `use_build_specific_project`? but I think you get the point.

Reply:
That seems okay - I do feel like a little more flexibility might be nice, but I can't think of a specific example off the top of my head. 

 I do think we should avoid falling back to `GOOGLE_CLOUD_PROJECT` if they've opted in - I'd rather throw an error to make it clear something is wrong.

Reply:
RE: using a `str` over `bool`, we may need to specify two different fall back projects for two different projects. With `true` we can only have one back up per build - by specifying and env var we'll have more freedom to have more fallback projects.

One example of this is currently the AutoML tests use a different project that the AutoML team maintains. We could specify one env var for the AutoML project, and still supply a different one for monitoring tests.

Reply:
It seems like automl tests are using "AUTOML_PROJECT_ID".

Can we just treat them as a different thing?

Reply:
We could, but it them means we have to expect the user to know to set `"AUTOML_PROJECT_ID"`. If we keep it wider, the samples could consistently refer to `"GOOGLE_CLOUD_PROJECT"` which would be more consistent (and hopefully intuitive) to other users, while the automl samples could take advantage of this noxfile feature to still customize the project.

Reply:
Now it's using str